services/api/src/rsu_ssm_srm.py

```diff
@@ -1,87 +1,117 @@
+                        "sigStatus"
+                    ]["signalStatusPackage"][0]["requester"]["role"],
+                    "status": doc["payload"]["data"]["status"]["signalStatus"][0][
+                        "sigStatus"
+                    ]["signalStatusPackage"][0]["status"],
+                    "type": doc["metadata"]["recordType"],
+                }
+            )
+        return 200, result
+    except Exception as e:
+        logging.error(f"SSM filter failed: {e}")
+        return 500, result
+
+
+def query_srm_data_mongo(result):
     end_date = datetime.now()
     end_utc = util.format_date_utc(end_date.isoformat())
     start_date = end_date - timedelta(days=1)
     start_utc = util.format_date_utc(start_date.isoformat())
-    client = bigquery.Client()
-    tablename = os.environ["SRM_DB_NAME"]
-
-    query = (
-        "SELECT rtdh_timestamp as time, srm.metadata.originIp as ip, "
-        f"srm.payload.data.requests.signalRequestPackage[ordinal(1)].request.requestID as request, "
-        f"srm.payload.data.requestor.type.role, "
-        f"srm.payload.data.requestor.position.position.latitude as lat, "
-        f"srm.payload.data.requestor.position.position.longitude as long, "
-        f"srm.metadata.recordType as type "
-        f'FROM `{tablename}` WHERE TIMESTAMP(rtdh_timestamp) >= "{start_utc}" '
-        f'AND TIMESTAMP(rtdh_timestamp) <= "{end_utc}" '
-        f"ORDER BY rtdh_timestamp ASC"
-    )
-
-    logging.info(f"Running query on table {tablename}")
-
-    query_job = client.query(query)
-
-    for row in query_job:
-        result.append(
-            {
-                "time": util.format_date_denver(row["time"].isoformat()),
-                "ip": row["ip"],
-                "requestId": row["request"],
-                "role": row["role"],
-                "lat": row["lat"],
-                "long": row["long"],
-                "type": row["type"],
-                "status": "N/A",
-            }
+
+    try:
+        client = MongoClient(os.getenv("MONGO_DB_URI"), serverSelectionTimeoutMS=5000)
+        db = client[os.getenv("MONGO_DB_NAME")]
+        collection = db[os.getenv("SRM_DB_NAME")]
+    except Exception as e:
+        logging.error(
+            f"Failed to connect to Mongo counts collection with error message: {e}"
         )
+        return [], 503
+
+    filter = {"recordGeneratedAt": {"$gte": start_utc, "$lte": end_utc}}
+    project = {
+        "recordGeneratedAt": 1,
+        "metadata.originIp": 1,
+        "metadata.recordType": 1,
+        "payload.data.requests.signalRequestPackage.request.requestID": 1,
+        "payload.data.requestor.type.role": 1,
+        "payload.data.requestor.position.position.latitude": 1,
+        "payload.data.requestor.position.position.longitude": 1,
+        "_id": 0,
+    }
 
-    return 200, result
+    logging.debug(f"Running filter on SRM mongoDB collection")
+
+    try:
+        for doc in collection.find(filter, project):
```

Comment:
Same comment as above, just to add context into our results and include a demo object we're looking at.

Reply:
Same for this one.